metadata/metadata.go

```diff
@@ -105,3 +149,53 @@ func safeCopy(destination map[string]string, key, val string) {
 		destination[key] = val
 	}
 }
+
+// CopyAnnotationsWithPrefixReplacement copies all annotations from a source object to a destination object where the key matches the specified sourcePrefix.
+// If destinationPrefix is different from sourcePrefix, the sourcePrefix will be replaced while performing the copy.
+func CopyAnnotationsWithPrefixReplacement(source, destination v1.Object, sourcePrefix, destinationPrefix string) error {
+	if _, err := GetAnnotationsWithPrefix(source, sourcePrefix); err != nil {
+		return err
+	}
+	if destination.GetAnnotations() == nil {
+		destination.SetAnnotations(make(map[string]string))
+	}
+	copyWithPrefixReplacement(source.GetAnnotations(), destination.GetAnnotations(), sourcePrefix, destinationPrefix)
+	return nil
+}
+
+// CopyAnnotationsByPrefix copies all annotations from a source object to a destination object where the key matches the specified sourcePrefix.
+func CopyAnnotationsByPrefix(source, destination v1.Object, prefix string) error {
+	if _, err := GetAnnotationsWithPrefix(source, prefix); err != nil {
+		return err
+	}
+	if destination.GetAnnotations() == nil {
+		destination.SetAnnotations(make(map[string]string))
+	}
+	copyByPrefix(source.GetAnnotations(), destination.GetAnnotations(), prefix)
+	return nil
+}
+
+// CopyLabelsWithPrefixReplacement copies all labels from a source object to a destination object where the key matches the specified sourcePrefix.
+// If destinationPrefix is different from sourcePrefix, the sourcePrefix will be replaced while performing the copy.
+func CopyLabelsWithPrefixReplacement(source, destination v1.Object, sourcePrefix, destinationPrefix string) error {
+	if _, err := GetLabelsWithPrefix(source, sourcePrefix); err != nil {
+		return err
+	}
+	if destination.GetLabels() == nil {
+		destination.SetLabels(make(map[string]string))
+	}
+	copyWithPrefixReplacement(source.GetLabels(), destination.GetLabels(), sourcePrefix, destinationPrefix)
+	return nil
+}
+
+// CopyLabelsByPrefix copies all labels from a source object to a destination object where the key matches the specified sourcePrefix.
+func CopyLabelsByPrefix(source, destination v1.Object, prefix string) error {
+	if _, err := GetLabelsWithPrefix(source, prefix); err != nil {
+		return err
+	}
+	if destination.GetLabels() == nil {
+		destination.SetLabels(make(map[string]string))
+	}
```

Comment:
I don't think you need the lines above, isn't it? `copyByPrefix` already filter by prefix and makes the copy.

The check in here should be something like:
```
if source == nil || destination == nil {
    return errors.New("objects cannot be nil")
}
```

Reply:
The same as the above one.